pantheon-bundle/src/main/java/com/redhat/pantheon/servlet/sitemap/SiteMapServlet.java

```diff
@@ -0,0 +1,172 @@
+
+            documentVariantResourceType = RESOURCE_TYPE_MODULEVARIANT;
+            documentVersionResourceType = RESOURCE_TYPE_MODULEVERSION;
+        } else if (request.getResource().getPath().startsWith("/api/sitemap/assembly.sitemap")) {
+            documentVariantResourceType = RESOURCE_TYPE_ASSEMBLYVARIANT;
+            documentVersionResourceType = RESOURCE_TYPE_ASSEMBLYVERSION;
+        } else {
+            log.warn("[" + SiteMapServlet.class.getSimpleName() + "] unhandled resource type: " + resource.getClass());
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            return;
+        }
+
+        List<Resource> documentAssets = new ArrayList<Resource>();
+        getAsset(resource, documentVariantResourceType, documentVersionResourceType, documentAssets);
+
+        if(documentAssets == null) {
+            response.setStatus(HttpServletResponse.SC_NOT_FOUND);
+            return;
+        }
+
+        response.setContentType(PantheonConstants.XML_MIME_TYPE);
+        response.setCharacterEncoding(PantheonConstants.UTF_8);
+
+        XMLOutputFactory outputFactory = XMLOutputFactory.newFactory();
+        try {
+            XMLStreamWriter stream = outputFactory.createXMLStreamWriter(response.getWriter());
+
+            stream.writeStartDocument(XML_DOCUMENT_VERSION);
+            stream.writeStartElement("", URL_SET, SITEMAP_NAMESPACE);
+            stream.writeNamespace("", SITEMAP_NAMESPACE);
+
+            documentAssets.forEach(r -> {
+                log.info("[" + SiteMapServlet.class.getSimpleName() + "] documentAssets r: " + r.getPath());
+            try {
+                writeXML(r, stream, request);
+            } catch (XMLStreamException e) {
+                e.printStackTrace();
+            }
+            });
+
+            stream.writeEndElement();
+            stream.writeEndDocument();
+
+        } catch (XMLStreamException e) {
+            throw new IOException(e);
+        }
+    }
+    private void writeXML(Resource resource, XMLStreamWriter xmlStream, SlingHttpServletRequest slingRequest)
+            throws XMLStreamException {
+        xmlStream.writeStartElement(SITEMAP_NAMESPACE, URL);
+
+        String locPath = resource.getPath();
+        DocumentVariant documentVariant = null;
+        Date dateModified = null;
+
+        // Process external url
+        if (System.getenv(PORTAL_URL) != null) {
+            if (resource.isResourceType(RESOURCE_TYPE_ASSEMBLYVARIANT)) {
+                documentVariant = resource.adaptTo(AssemblyVariant.class);
+            } else if (resource.isResourceType(RESOURCE_TYPE_MODULEVARIANT)) {
+                documentVariant = resource.adaptTo(ModuleVariant.class);
+            }
+            if (documentVariant != null && documentVariant.released().isPresent()) {
+                locPath = new CustomerPortalUrlUuidProvider().generateUrlString(documentVariant);
```

Comment:
You could probably re-use the same CustomerPortalUrlUuidProvider instance, but I don't think creating a new one is particularly expensive either, so this is pretty minor.

Reply:
updated